spec/models/spree/variant_spec.rb

```diff
@@ -166,7 +166,8 @@ module Spree
       let(:variant) { create(:variant) }
 
       it "refreshes the products cache on save" do
-        expect(OpenFoodNetwork::ProductsCache).to receive(:variant_changed).with(variant)
+        # When creating the variant both the Variant and StockItem callbacks get executed
```

Comment:
I prefer to optimize these things in a separate PR. From my experience, you can end up in a black hole very easily when dealing with them and we need this fix ASAP.

What I have in mind is using something like

```
def refresh_products_cache
  unless variant.id_was == nil # at this point we know we're creating a new variant
    # refresh logic
  end
end
```

Or perhaps simply an `after_update` and an `after_destroy` instead of this broad `after_save`.